app/models/manager_refresh/target_collection.rb

```diff
@@ -0,0 +1,76 @@
+module ManagerRefresh
+  class TargetCollection
+    attr_reader :targets
+
+    delegate :<<, :to => :targets
+
+    # @param manager [ManageIQ::Providers::BaseManager] The Manager owning the TargetCollection
+    # @param event [EmsEvent] AnEmsEvent associated with the TargetCollection
+    # @param targets [Array] An Array of ManagerRefresh::Target objects or ApplicationRecord objects
+    def initialize(manager: nil, manager_id: nil, event: nil, targets: [])
+      @manager    = manager
+      @manager_id = manager_id
+      @event      = event
+      @targets    = targets
+    end
+
+    # @param association [Symbol] An existing association on Manager, that lists objects represented by a Target, naming
+    #                             should be the same of association of a counterpart InventoryCollection object
+    # @param manager_ref [Hash] A Hash that can be used to find_by on a given association and returning a unique object.
+    #                           The keys should be the same as the keys of the counterpart InventoryObject
+    # @param manager [ManageIQ::Providers::BaseManager] The Manager owning the Target
+    # @param manager_id [Integer] A primary key of the Manager owning the Target
+    # @param event_id [Integer] A primary key of the EmsEvent associated with the Target
+    # @param options [Hash] A free form options hash
+    def add_target!(association:, manager_ref:, manager: nil, manager_id: nil, event_id: nil, options: {})
```

Comment:
I would not add a `!` here. `add_` clearly indicates that you are modifying it. It's more like `Array#push`